File: rust/src/models/message_attempt_failed_data.rs
// this file is @generated
use serde::{Deserialize, Serialize};

#[derive(Clone, Debug, Default, PartialEq, Deserialize, Serialize)]
pub struct MessageAttemptFailedData {
    /// The MessageAttempt's ID.
    pub id: String,

    #[serde(rename = "responseStatusCode")]
    pub response_status_code: i16,

    pub timestamp: String,
}

impl MessageAttemptFailedData {
    pub fn new(id: String, response_status_code: i16, timestamp: String) -> Self {
        Self {
            id,
            response_status_code,
            timestamp,
        }
    }
}

File: javascript/src/models/messageAttemptFailedData.ts
// this file is @generated

export interface MessageAttemptFailedData {
  /** The MessageAttempt's ID. */
  id: string;
  responseStatusCode: number;
  timestamp: Date;
}

export const MessageAttemptFailedDataSerializer = {
  _fromJsonObject(object: any): MessageAttemptFailedData {
    return {
      id: object["id"],
      responseStatusCode: object["responseStatusCode"],
      timestamp: new Date(object["timestamp"]),
    };
  },

  _toJsonObject(self: MessageAttemptFailedData): any {
    return {
      id: self.id,
      responseStatusCode: self.responseStatusCode,
      timestamp: self.timestamp,
    };
  },
};
